### PR TITLE
MP-4394 Migrate IntelliJ Plugin Verifier to Gradle Kotlin DSL

### PR DESCRIPTION
**Pull request by novotnyr**
Migrate IntelliJ Plugin Verifier to Gradle Kotlin DSL

- bump Gradle to 8.1.1
- migrate build scripts to Gradle
- update Gradle API that got broken between 7.x and 8.1.x

### CODE REVIEW

File: intellij-plugin-verifier/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,138 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    `maven-publish`
+    id("org.jetbrains.kotlin.jvm") version "1.8.10"
```
Comment (YannCebron) [inline intellij-plugin-verifier/build.gradle.kts:5]: why not latest 1.8.21?

File: intellij-plugin-verifier/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+val kotlin_version by extra("1.8.20")
```
Comment (YannCebron) [inline intellij-plugin-verifier/build.gradle.kts:8]: 1.8.21?

File: intellij-plugin-verifier/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    implementation("com.intellij:annotations:12.0")
+    implementation("commons-io:commons-io:2.10.0")
+    implementation("com.google.code.gson:gson:2.10.1")
+  }
+
+  repositories {
+    mavenCentral()
+    mavenLocal()
+    maven { url = uri("https://www.jetbrains.com/intellij-repository/releases") }
+    maven { url = uri("https://packages.jetbrains.team/maven/p/ij/intellij-dependencies") }
```
Comment (YannCebron) [inline intellij-plugin-verifier/build.gradle.kts:72]: https://cache-redirector.jetbrains.com/intellij-dependencies instead?

File: intellij-plugin-verifier/verifier-test/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+val afterIdeaBuildDir = project("after-idea").buildDir
+val additionalAfterIdeaBuildDir = project("additional-after-idea").buildDir
+
+/**
+ * Creates resources.jar file with brokenPlugins.txt file inside.
+ */
+
+val prepareResourcesJar by tasks.registering(Jar::class) {
+  dependsOn(  tasks.getByPath("after-idea:processResources"))
```
Comment (YannCebron) [inline intellij-plugin-verifier/verifier-test/build.gradle.kts:26]: reformat

File: intellij-plugin-verifier/build.gradle.kts [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+}
+
+val kotlin_version by extra("1.8.20")
+
+val projectVersion: String by extra {
+  project.properties.getOrDefault("verifierProjectVersion", "dev").toString()
+}
+
+val isDevMode = projectVersion == "dev"
+var ijStructureVersion = "dev"
```
Comment (LChernigovskaya) [inline intellij-plugin-verifier/build.gradle.kts:15]:
```
val intellijStructureVersion = if (!isDevMode) {
  if (project.hasProperty("intellijStructureVersion")) {
   project.properties["intellijStructureVersion"].toString()
  } else {
    throw RuntimeException("Version of intellij-structure library to be used is not specified via 'intellijStructureVersion'")
  }
  else {
  "dev"
  }
}
```

File: intellij-plugin-verifier/verifier-core/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,8 @@
+repositories {
```
Comment (LChernigovskaya) [inline intellij-plugin-verifier/verifier-core/build.gradle.kts:1]: Will it be got from the root gradle file?

Comment (novotnyr) [reply intellij-plugin-verifier/verifier-core/build.gradle.kts:1]: Yes, this is inherited from `projects`. I will remove that.

**Pull request merged**